### PR TITLE
remove sg from private lbs

### PR DESCRIPTION
**Pull request by zpatrick**
closes https://github.com/quintilesims/layer0/issues/194

### CODE REVIEW

File: api/backend/ecs/load_balancer_manager.go [hunk 1]
```diff
@@ -304,17 +303,23 @@ func (e *ECSLoadBalancerManager) createLoadBalancer(
 		return err
 	}
 
-	securityGroup, err := e.upsertSecurityGroup(ecsLoadBalancerID, ports)
-	if err != nil {
-		return err
+	// only public load balancers get an additional security group
```
Comment (diemonster) [inline api/backend/ecs/load_balancer_manager.go]: are there any side effects here for private load balancers when envs are linked?

Comment (zpatrick) [inline api/backend/ecs/load_balancer_manager.go]: Not that I've seen so far. I ran the system tests to verify:
- Public->Private communication without environment linking fails
- Public->Private communication with linking succeeds 

I also spun up a consul cluster behind a private lb and it was able to communicate to itself as expected.

Comment (diemonster) [inline api/backend/ecs/load_balancer_manager.go]: > I also spun up a consul cluster behind a private lb and it was able to communicate to itself as expected.

the main case I'm worried about here is a consul cluster in it's own environment, with a separate environment's service(s) trying to hit the consul cluster's private load balancer. If the private load balancer doesn't get the 'Access all' SG, I'm unclear on how other envs will be able to hit it.

Comment (zpatrick) [inline api/backend/ecs/load_balancer_manager.go]: that's what was verified with: 
```
* Public->Private communication without environment linking fails
* Public->Private communication with linking succeeds
```

Comment (zpatrick) [inline api/backend/ecs/load_balancer_manager.go]: and it does get the 'Access all' SG from the environment

Comment (diemonster) [inline api/backend/ecs/load_balancer_manager.go]: Awesome, ok. 👍

**Pull request merged**